web/app/scripts/resources/recordexports-service.js

```diff
@@ -0,0 +1,78 @@
+(function () {
+    'use strict';
+
+    /**
+     * Service for creating and downloading record exports.
+     * exportCSV takes a tilekey and initiates an export then polls for the result, returning
+     * a promise that either resolves with a download URL or rejects with an error message.
+     */
+
+    /* ngInject */
+    function RecordExports($resource, $q, $interval, WebConfig) {
+        var pollingInterval;
+        var POLLING_INTERVAL_MS = 1500;
+        var MAX_POLLING_TIME_S = 100;
+
+        var Exports = $resource(WebConfig.api.hostname + '/api/csv-export/:id/', {id: '@uuid'}, {
+            create: {
+                method: 'POST'
+            },
+            get: {
+                method: 'GET'
+            },
+        });
+
+        var module = {
+            exportCSV: exportCSV,
+            cancelPolling: cancelPolling,
+        };
+        return module;
+
+        function exportCSV(tilekey) {
+            cancelPolling();
+            var deferred = $q.defer();
+
+            Exports.create({ tilekey: tilekey },
+                function (result) {
+                    var taskID = result.taskid;
+                    pollingInterval = $interval(
+                        function () {
+                            Exports.get({ id: taskID }).$promise.then(function (response) {
+                                switch (response.status) {
+                                    case 'PENDING':
```

Comment:
Maybe this is excessively pedantic, but an explicit `break` here might be good rather than letting it fall through, in case somebody comes back later and makes changes without recognizing the fall-through behavior.